### DISCUSSION TITLE
Postgres instead of RocksDB?

### DISCUSSION THREAD

**Discussion by mispp** [stalwartlabs/jmap-server#47] [Ideas]
Hello,

is it planned somewhere to support postgres for message store (blobs on filesystem with references in db)?
Given that smtp-server uses postgres, this might also reuse the same db.
Plus, full text search might work well for mail content.

Thanks

**Comment by mdecimus** [comment]
The next version of the JMAP server will support either SQLite or FoundationDB. Adding PostgreSQL support is possible but would not be very performant as Stalwart JMAP implements the query logic internally and for that reason it requires a key/value store.
